roles/openshift_inventory/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- no_log: True  # keep task descriptions legible
+  block:
+
+  - name: Standardize on latest variable names
+    set_fact:
+      # goal is to deprecate deployment_type in favor of openshift_deployment_type.
+      # both will be accepted for now, but code should refer to the new name.
+      # TODO: once this is well-documented, add deprecation notice if using old name.
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+      openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+  - name: Normalize openshift_release
+    set_fact:
+      # normalize release if provided, e.g. "v3.5" => "3.5"
+      openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
+    when: openshift_release is defined
+
+- name: Ensure a valid deployment type has been given.
+  # this variable is required; complain early and clearly if it is invalid.
+  when: |
+    openshift_deployment_type is not defined
```

Comment:
Good point, I was thinking `None` did not count as defined but python/jinja2 do not see it that way. Can remove this line.